src/com/tony/server/Authenticator.java

```diff
@@ -0,0 +1,56 @@
+package com.tony.server;
+
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashMap;
+
+public class Authenticator {
+    private static final int ENCODED_USER_INFO = 1;
+    private ArrayList<Request> authenticationList;
+    private ArrayList<String> authenticatedUsers;
+
+    public Authenticator() {
+        this.authenticationList = new ArrayList<>();
+        this.authenticatedUsers = new ArrayList<>();
+    }
+
+    public void addToAuthenticationList(Request request) {
+       authenticationList.add(request);
+    }
+
+    public void addToAuthenticatedUsers(String userInfo) {
+        authenticatedUsers.add(userInfo);
+    }
+
+    public ArrayList<Request> getAuthenticationList() {
+        return authenticationList;
+    }
+
+    public ArrayList<String> getAuthenticatedUsers() {
+        return authenticatedUsers;
+    }
+
+    public boolean requiresAuthentication(Request request) {
+        return authenticationList.contains(request);
+    }
+
+    public boolean isAuthenticated(Request request) {
+        HashMap headers = request.getHeaders();
+        if(headers.containsKey("Authorization")){
+            String authorizationValue = request.getHeaders().get("Authorization");
+            String encodedUserInfo = authorizationValue.split(" ")[ENCODED_USER_INFO];
+            return authenticatedUsers.contains(decodeUserInfo(encodedUserInfo));
+        }
+        return false;
+    }
+
+    public static String decodeUserInfo(String encodedUserInfo) {
+        String decodedUserInfo = "";
+        try{
+           decodedUserInfo = new String(Base64.getDecoder().decode(encodedUserInfo));
+        } catch(Exception e){
+           return "Error in decoding.";
```

Comment:
Do you really want to be returning this "message" as the return value from this method? This seems like a great place to put your logger to use and log some useful information about what error occurred. Which brings me to my next point, you shouldn't be catching exceptions with `catch(Exception...`, as it hides which exception was actually thrown, and doesn't force you to understand what is happening inside of the catch that could blow up. 

In this particular case it looks like the only thing that could throw an exception is [Base64.Decoder#decode](https://docs.oracle.com/javase/8/docs/api/java/util/Base64.Decoder.html#decode-java.lang.String-).